tests/device_tests/test_msg_signtx.py

```diff
@@ -221,13 +240,59 @@ def test_testnet_fee_too_high(self, client):
 
         assert (
             tx_hash(serialized_tx).hex()
-            == "c669527e0d80dc645925f6965e1622e71fa5ca51e284442c4620c1ade7a76c63"
+            == "54fd5e9b65b8acc10144c1e78ea9720df7606d7d4a543e4c547ecd45b2ae226b"
+        )
+
+    @pytest.mark.skip_t1
+    def test_testnet_fee_high_hardfail(self, client):
+        # tx: 6f90f3c7cbec2258b0971056ef3fe34128dbde30daa9c0639a898f9977299d54
+        # input 1: 10.00000000 BTC
+        inp1 = messages.TxInputType(
+            address_n=parse_path("44'/1'/0'/0/0"),
+            # amount=1000000000,
+            prev_hash=TXHASH_6f90f3,
+            prev_index=1,
+        )
+
+        out1 = messages.TxOutputType(
+            address="mfiGQVPcRcaEvQPYDErR34DcCovtxYvUUV",
+            amount=1000000000 - 500000000 - 80000000,
+            script_type=messages.OutputScriptType.PAYTOADDRESS,
+        )
+
+        out2 = messages.TxOutputType(
+            address_n=parse_path("44'/1'/0'/1/0"),
+            amount=500000000,
+            script_type=messages.OutputScriptType.PAYTOADDRESS,
+        )
+
+        with pytest.raises(TrezorFailure, match="fee is unexpectedly large"):
+            btc.sign_tx(
+                client, "Testnet", [inp1], [out1, out2], prev_txes=TX_CACHE_TESTNET
+            )
+
+        # set SafetyCheckLevel to Prompt and try again
+        with client:
+            checker = CheckCodeSent(client, B.FeeOverThreshold)
+            client.set_input_flow(checker)
+
+            device.apply_settings(
```

Comment:
please move the `device.apply_settings()` call(s) out of the `with client` block(s), so that their responses are not part of the with-block

Reply:
Done in 52b45027.